include/residuals.h

```diff
@@ -8,6 +8,13 @@
 
 namespace GLnexus {
 
+// Dataset and the records it has for a particular site
+struct DatasetSiteInfo {
+    std::string name;
+    std::shared_ptr<const bcf_hdr_t> header;
+    std::vector<std::shared_ptr<bcf1_t>> records;
+};
+
 class Residuals {
```

Comment:
Is the `Residuals` class still necessary? It seems like it's just a lambda, in that it has one member function and no mutable state. Below it gets allocated and passed around in a `shared_ptr`, and is only invoked in one place. Could this be simplified to just a function call?